server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -246,10 +246,10 @@ protected void doStop() {
 
         try {
             CloseableChannel.closeChannels(new ArrayList<>(httpChannels), true);
+            assert httpChannels.isEmpty() : "all channels should have been closed but saw [" + httpChannels + "]";
```

Comment:
I think this is ok, `CloseableChannel.closeChannels(..., true)` adds a new close listener to all the channels and I am assuming that the listeners definitely complete in the order in which they were added.

Reply:
Welp https://gradle-enterprise.elastic.co/s/aaolqwri242tm seems to disagree 😕

Reply:
Unfortunately turns out this doesn't work because in case the channel is already when the listener is added it will potentially run before queued listeners. Looking into an alternative check ...

Reply:
hmpf that required some extra complexity https://github.com/elastic/elasticsearch/pull/76257/commits/f4d5454f1c62371115958bf07695118990b24b59 but it may be well worth it, we shouldn't shut down the even loop before all connections have been closed to begin with I think.